File: pkg/dev_compiler/test/codegen_expected/lib/convert/close_test.js
dart_library.library('lib/convert/close_test', null, /* Imports */[
  'dart_sdk',
  'expect'
], function load__close_test(exports, dart_sdk, expect) {
  'use strict';
  const core = dart_sdk.core;
  const _interceptors = dart_sdk._interceptors;
  const convert = dart_sdk.convert;
  const dart = dart_sdk.dart;
  const dartx = dart_sdk.dartx;
  const expect$ = expect.expect;
  const close_test = Object.create(null);
  let JSArrayOfint = () => (JSArrayOfint = dart.constFn(_interceptors.JSArray$(core.int)))();
  let ListOfint = () => (ListOfint = dart.constFn(core.List$(core.int)))();
  let SinkOfListOfint = () => (SinkOfListOfint = dart.constFn(core.Sink$(ListOfint())))();
  let VoidTodynamic = () => (VoidTodynamic = dart.constFn(dart.definiteFunctionType(dart.dynamic, [])))();
  close_test.MySink = class MySink extends core.Object {
    new() {
      this.accumulated = JSArrayOfint().of([]);
      this.isClosed = false;
    }
    add(list) {
      this.accumulated[dartx.addAll](list);
      list[dartx.length];
    }
    close() {
      this.isClosed = true;
    }
  };
  close_test.MySink[dart.implements] = () => [SinkOfListOfint()];
  dart.setSignature(close_test.MySink, {
    methods: () => ({
      add: dart.definiteFunctionType(dart.void, [core.List$(core.int)]),
      close: dart.definiteFunctionType(dart.void, [])
    })
  });
  close_test.main = function() {
    let mySink = new close_test.MySink();
    let byteSink = convert.ByteConversionSink.from(mySink);
    byteSink.add(JSArrayOfint().of([1, 2, 3]));
    byteSink.close();
    expect$.Expect.listEquals(JSArrayOfint().of([1, 2, 3]), mySink.accumulated);
    expect$.Expect.isTrue(mySink.isClosed);
  };
  dart.fn(close_test.main, VoidTodynamic());
  // Exports:
  exports.close_test = close_test;
});

File: tests/lib_strong/convert/close_test.dart
// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
// for details. All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.

import "package:expect/expect.dart";
import "dart:convert";

class MySink implements Sink<List<int>> {
  List<int> accumulated = <int>[];
  bool isClosed = false;

  add(List<int> list) {
    accumulated.addAll(list);
    list.length;
  }

  close() {
    isClosed = true;
  }
}

main() {
  var mySink = new MySink();
  var byteSink = new ByteConversionSink.from(mySink);
  byteSink.add([1, 2, 3]);
  byteSink.close();
  Expect.listEquals([1, 2, 3], mySink.accumulated);
  Expect.isTrue(mySink.isClosed);
}
